src/order/bounded_lattice.lean

```diff
@@ -109,35 +108,38 @@ lemma ne.lt_top' (h : ⊤ ≠ a) : a < ⊤ := h.symm.lt_top
 
 end order_top
 
-lemma strict_mono.maximal_preimage_top [linear_order α] [order_top β]
+lemma strict_mono.maximal_preimage_top [linear_order α] [preorder β] [order_top β]
   {f : α → β} (H : strict_mono f) {a} (h_top : f a = ⊤) (x : α) :
   x ≤ a :=
 H.maximal_of_maximal_image (λ p, by { rw h_top, exact le_top }) x
 
-theorem order_top.ext_top {α} {A B : order_top α}
+theorem order_top.ext_top {α} [partial_order α] (A B : order_top α)
   (H : ∀ x y : α, (by haveI := A; exact x ≤ y) ↔ x ≤ y) :
   (by haveI := A; exact ⊤ : α) = ⊤ :=
```

Comment:
I don't think this `by haveI` makes sense any more, since there's only one `has_le` instance around.

Are you sure this wasn't supposed to have two different `partial_order` instances too?

Reply:
Something like (untested):
```suggestion
theorem order_top.ext_top {α} {hA : partial_order α} (A : order_top α) {hB : partial_order α} (B : order_top α)
  (H : ∀ x y : α, (by haveI := A; exact x ≤ y) ↔ x ≤ y) :
  (by haveI := A; exact ⊤ : α) = ⊤ :=
```

Reply:
You'll have to make sure `B` picks up `hB` and not `hA` using the infoview.